tt_eager/tensor/types.hpp

```diff
@@ -212,6 +212,15 @@ class Shape {
         return std::forward_as_tuple(this->rank_, this->dimensions_, this->padding_);
     }
     friend std::ostream &operator<<(std::ostream &os, const Shape &shape);
+
+    std::vector<uint32_t> to_vector() const {
```

Comment:
We avoid vectors, because they require heap allocation which leads to slowdowns.
Do we absolutely need it?

 FYI @dmakoviichuk-tt